crates/bls_utils/src/verification.rs

```diff
@@ -71,16 +81,25 @@ pub fn verify_seed_exchange_commitment(
     let commitment = &seed_exchange.commitment;
     let shared_secret = &seed_exchange.shared_secret;
 
-    if !bls_verify(
-        &G1Affine::from_compressed(&commitment.pubkey)
-            .into_option()
-            .unwrap(),
-        &G2Affine::from_compressed(&commitment.signature)
-            .into_option()
-            .unwrap(),
-        &commitment.hash,
+    let g1_pubkey = to_g1_affine(&commitment.pubkey);
+
+    let g2_sig = match G2Affine::from_compressed(&commitment.signature).into_option() {
+        Some(g2_sig) => g2_sig,
```

Comment:
I was going to suggest the same, as I was looking at the previous version of this code locally. Great minds!